src/test/java/seedu/address/logic/commands/ClientNoteEditCommandTest.java

```diff
@@ -0,0 +1,124 @@
+
+    private Model model;
+
+    @BeforeEach
+    public void setUp() {
+        model = new ModelManager(getTypicalTbmManager(), new UserPrefs());
+    }
+
+    @Test
+    public void execute_validClientIdxValidNoteIdx_generatesClientNoteEditCommandSuccessfully() {
+        Index clientIdx = Index.fromOneBased(1);
+        Index clientNoteIdx = Index.fromOneBased(1);
+        Note clientNote1 = new Note(NOTE_CONTENT_1);
+        Note clientNote2 = new Note(NOTE_CONTENT_2);
+        Note newEditNote = new Note("dummy note to edit previous note");
+        Model newModel = new ModelManager();
+        Client client1 = new ClientBuilder().withName("client1").build();
+        newModel.addClient(client1);
+        newModel.addClientNote(client1, clientNote1);
+        newModel.addClientNote(client1, clientNote2);
+
+        Model expectedModel = new ModelManager();
+        Client client1Copy = new ClientBuilder().withName("client1").build();
+        expectedModel.addClient(client1Copy);
+        expectedModel.addClientNote(client1Copy, newEditNote);
+        expectedModel.addClientNote(client1Copy, clientNote2);
+
+        CommandResult expectedResult = new CommandResult(ClientNoteEditCommand.MESSAGE_EDIT_CLIENT_NOTE_SUCCESS);
+        ClientNoteEditCommand clientNoteEditCommand = new ClientNoteEditCommand(clientIdx, clientNoteIdx, newEditNote);
+        assertCommandSuccess(clientNoteEditCommand, newModel, expectedResult, expectedModel);
+    }
+
+    @Test
+    public void execute_validClientIdxValidNoteIdxNewTaggedNote_preservesTagHistory() {
+        Tag oldTag = new Tag("historicalTag");
+        Tag newTag = new Tag("newTag");
+        Set<Tag> newTagSet = new HashSet<>();
+        newTagSet.add(newTag);
+        Set<Tag> tagHistory = new HashSet<>();
+        tagHistory.add(oldTag);
+        Set<Tag> expectedTags = new HashSet<>(); // preserves old tags and new tags
+        expectedTags.add(oldTag);
+        expectedTags.add(newTag);
+        Index clientIdx = Index.fromOneBased(1);
+        Index clientNoteIdx = Index.fromOneBased(1);
+        Note oldClientNote = new Note(NOTE_CONTENT_1);
+        oldClientNote.setTags(tagHistory);
+        Note clientNote2 = new Note(NOTE_CONTENT_2);
+        Note newNote = new Note("dummy note to edit previous note");
+        newNote.setTags(newTagSet);
+        Model newModel = new ModelManager();
+        Client client1 = new ClientBuilder().withName("client1").build();
+        newModel.addClient(client1);
+        newModel.addClientNote(client1, oldClientNote);
+        newModel.addClientNote(client1, clientNote2);
+
+        Model expectedModel = new ModelManager();
+        Client client1Copy = new ClientBuilder().withName("client1").build();
+        expectedModel.addClient(client1Copy);
+        newNote.setTags(expectedTags);
+        expectedModel.addClientNote(client1Copy, newNote);
+        expectedModel.addClientNote(client1Copy, clientNote2);
+
```

Comment:
trying to refactor this test rn. The logic behind the test should be correct in that it's trying to ensure that editing a note preserves the note's tagging history.